src/Core/Grid/Query/AliasQueryBuilder.php

```diff
@@ -0,0 +1,124 @@
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Query;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+class AliasQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var DoctrineSearchCriteriaApplicatorInterface
+     */
+    private $searchCriteriaApplicator;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
+     */
+    public function __construct(Connection $connection, $dbPrefix, DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator)
+    {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->searchCriteriaApplicator = $searchCriteriaApplicator;
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return QueryBuilder
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria): QueryBuilder
+    {
+        $builder = $this->getAliasQueryBuilder($searchCriteria);
+
+        $builder
+            ->select('a.id_alias, a.alias, a.search, a.active, COUNT(a.id_alias) as members')
+            ->groupBy('a.id_alias')
+        ;
+
+        $this->searchCriteriaApplicator
+            ->applySorting($searchCriteria, $builder)
+            ->applyPagination($searchCriteria, $builder);
+
+        return $builder;
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return QueryBuilder
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria): QueryBuilder
+    {
+        return $this->getAliasQueryBuilder($searchCriteria)->select('COUNT(a.id_alias)');
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return QueryBuilder
+     */
+    private function getAliasQueryBuilder(SearchCriteriaInterface $searchCriteria): QueryBuilder
+    {
+        $builder = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'alias', 'a')
+        ;
+
+        $this->applyFilters($builder, $searchCriteria);
+
+        return $builder;
+    }
+
+    /**
+     * @param QueryBuilder $builder
+     * @param SearchCriteriaInterface $searchCriteria
+     */
+    private function applyFilters(QueryBuilder $builder, SearchCriteriaInterface $searchCriteria): void
+    {
+        $allowedFiltersMap = [
+            'id_alias' => 'a.id_alias',
+            'alias' => 'a.alias',
+            'search' => 'a.search',
+            'active' => 'a.active',
+        ];
+
+        foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
+            if (!array_key_exists($filterName, $allowedFiltersMap)) {
+                continue;
+            }
+
+            $builder
+                ->andWhere($allowedFiltersMap[$filterName] . ' LIKE :' . $filterName)
```

Comment:
id_alias field should be a strict search. (this is not a strict convention and we have some mixed behaviors, but I think thats the way we are going now - id fields filters uses strict search or a range input, but lets keep it simple for now.)
example here:
https://github.com/PrestaShop/PrestaShop/pull/31802/files#diff-4e2bb3656e21bd0da6e30a111a6dc99f42dd1dc487431714d80e7cfbff9a4f61R166